apisix/balancer.lua

```diff
@@ -136,8 +136,18 @@ end
 -- set_balancer_opts will be called in balancer phase and before any tries
 local function set_balancer_opts(ctx)
```

Comment:
We can pass route directly:
https://github.com/apache/apisix/blob/09e298302cbc8514d8240c300974746c9f85845c/apisix/balancer.lua#L247